tests/pages/tap.test.js

```diff
@@ -129,7 +129,9 @@ test('Correct error appears when I submit a duplicate voucher', async t => {
         .click(submitVoucher)
     ;
     const errorMessage = await el('.message').innerText;
-    expect(errorMessage).to.contain('It looks like this code has already been added, please double check and try again. If you are still unable to add the voucher code, don\'t worry - you will still receive payment if you send it in with your other vouchers.');
+    expect(errorMessage).to.contain("It looks like the code (:code) has been used already, please double check and try again. "
+    + "If you are still unable to add the voucher code, don't worry - mark it as \"unrecorded\", "
```

Comment:
can't use "." to concatenate JS. Js uses "+" as it's string concat operator. is that what you mean?